compiler/src/iree/compiler/API/Internal/IREEGPUDialectCAPI.cpp

```diff
@@ -148,6 +148,9 @@ MlirTypeID ireeGPUVirtualMMAIntrinsicAttrGetTypeID() {
       mlir::iree_compiler::IREE::GPU::VirtualMMAIntrinsicAttr::getTypeID());
 }
 
+static_assert(std::is_same_v<mma_intrinsic_enum_t, uint32_t>,
+              "mma_intrinsic_enum_t must be uint32_t");
```

Comment:
Instead of this, I think we should update the two enums below. Our code is correct when `mma_intrinsic_enum_t` matches the underlying type of both mma enums